### PR TITLE
Symfony 6

### PR DESCRIPTION
**Pull request by jongotlin**
| Q             | A
| ------------- | ---
| Bug fix?      |no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | 
| License       | MIT

Not 100% sure this is all it takes to make it work in Symfony 6. At least main functionality is working in my Symfony 6 app.

### CODE REVIEW

File: EventListener/MainRequestTrait.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+namespace CoopTilleuls\ForgotPasswordBundle\EventListener;
+
+use Symfony\Component\HttpKernel\Event\KernelEvent;
+
+/**
+ * @author Jon Gotlin <jon@jon.se>
+ */
+trait MainRequestTrait
+{
+    protected function isMainRequest(KernelEvent $event): bool
```
Comment (vincentchalamon) [inline EventListener/MainRequestTrait.php:23]:
```suggestion
    private function isMainRequest(KernelEvent $event): bool
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -22,13 +22,18 @@ jobs:
         symfony:
           - '4.4.*'
           - '5.3.*'
```
Comment (vincentchalamon) [inline .github/workflows/ci.yml:24]:
```suggestion
          - '5.4.*'
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -22,13 +22,18 @@ jobs:
...
+          - '6.0.*'
         include:
           - php: '8.0'
             symfony: '5.3.*'
```
Comment (vincentchalamon) [inline .github/workflows/ci.yml:28]:
```suggestion
            symfony: '5.4.*'
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -22,13 +22,18 @@ jobs:
...
             bootable: true
```
Comment (vincentchalamon) [inline .github/workflows/ci.yml:29]:
```suggestion
            bootable: true
            quality: true
```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -65,6 +70,7 @@ jobs:
           composer req --ignore-platform-reqs tilleuls/forgot-password-bundle:dev-${GITHUB_REF#refs/heads/}
       - name: Run php-cs-fixer tests
         run: php-cs-fixer fix --diff --dry-run
+        if: matrix.php >= '7.4'
```
Comment (vincentchalamon) [inline .github/workflows/ci.yml:73]:
```suggestion
        if: matrix.quality
```

**Review by vincentchalamon — CHANGES_REQUESTED** [review]
Great work, thank you very much @jongotlin!
Some minor changes, and the CI is failing, but then I think it'll be ok :-)

**Comment by vincentchalamon** [comment]
You gonna need to also update the DocumentationNormalizer and the AppKernel files:
> Method "Symfony\Component\Serializer\Normalizer\NormalizerInterface::supportsNormalization()" might add "bool" as a native return type declaration in the future. Do the same in implementation "CoopTilleuls\ForgotPasswordBundle\Bridge\ApiPlatform\Serializer\DocumentationNormalizer" now to avoid errors or add an explicit @return annotation to suppress this message.

> PHP Fatal error:  Declaration of AppKernel::getProjectDir() must be compatible with Symfony\Component\HttpKernel\Kernel::getProjectDir(): string in /home/runner/work/CoopTilleulsForgotPasswordBundle/CoopTilleulsForgotPasswordBundle/features/app/AppKernel.php on line 43

**Comment by jongotlin** [comment]
Thank you for your review @vincentchalamon! Couldn't get the tests pass on sf 6. Can you help out? Otherwise I'll dig into it some more in the next couple of days.

**Comment by jongotlin** [comment]
Got stuck with this one and would appreciate some help to get the tests pass with Symfony 6.

**Comment by vincentchalamon** [comment]
@jongotlin I've commited some modifications on AppKernel to unstuck this PR. Could you have a look at the failing tests please? Should be easier now

**Comment by jongotlin** [comment]
Thank you @vincentchalamon! Fixed the remaining stuff.

**Review by vincentchalamon — APPROVED** [review]

**Comment by vincentchalamon** [comment]
Great job, thank you very much @jongotlin !

**Pull request merged**